File: 05072024/FrontEnd/src/app/dto/Categoria.ts
export class Categoria {
  private _id: number;
  private _nomeCategoria: string;

  constructor(id: number, nomeCategoria: string) {
    this._id = id;
    this._nomeCategoria = nomeCategoria;
  }

  public get id(): number {
    return this._id;
  }

  public get nomeCategoria(): string {
    return this._nomeCategoria;
  }

  public set id(id: number) {
    this._id = id;
  }

  public set nomeCategoria(nomeCategoria: string) {
    this._nomeCategoria = nomeCategoria;
  }
}

File: 28062024/restFinalFedericoRifugiato/src/main/java/com/federicorifugiato/model/Categoria.java
package com.federicorifugiato.model;

import java.util.ArrayList;
import java.util.List;


import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "categoria")
public class Categoria {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)    
    @Column(name = "ID_CA")
    private int id;
    
    @Column(name = "Nome_Categoria")
    private String nomeCategoria;
    
    @OneToMany(mappedBy = "categoria", cascade = CascadeType.REFRESH, orphanRemoval = true, fetch = FetchType.EAGER)
    private List<Corso> corsi = new ArrayList<>();

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNomeCategoria() {
		return nomeCategoria;
	}

	public void setNomeCategoria(String nomeCategoria) {
		this.nomeCategoria = nomeCategoria;
	}

	public List<Corso> getCorsi() {
		return corsi;
	}

	public void setCorsi(List<Corso> corsi) {
		this.corsi = corsi;
	}
    
}
